Main/Controllers/CollectionsController.cs

```diff
@@ -47,5 +47,64 @@ public IActionResult AddCollection([FromBody] CollectionCreationData data)
       _dbContext.SaveChanges();
       return Ok();
     }
+
+    [HttpDelete("{id}")]
+    public IActionResult DeleteCollection(int id)
+    {
+      var userId = _userProvider.GetCurrentUserId();
+      var collection = _dbContext.Collection.SingleOrDefault(c => c.Id == id);
+      if (collection == null)
+        return NotFound();
+      if (collection.OwnerId == userId)
+      {
+        using (var dbContextTransaction = _dbContext.Database.BeginTransaction())
+        {
```

Comment:
Also you can do "using var dbcontexttransaction = ...;" in C#8. Helps with eliminating nesting